src/Build.UnitTests/Evaluation/Evaluator_Tests.cs

```diff
@@ -4528,6 +4483,76 @@ public void VerifyMSBuildLogsAMessageWhenLocalPropertyCannotOverrideValueOfGloba
                 ResourceUtilities.FormatResourceStringStripCodeAndKeyword("OM_GlobalProperty", "Foo"));
         }
 
+        [Fact]
+        public void VerifyPropertyTrackingLogging()
+        {
+           string testtargets = ObjectModelHelpers.CleanupFileContents(@"
+                                <Project>
+                                     <PropertyGroup>
+                                         <Prop>$(DOES_NOT_EXIST)</Prop>
+                                         <EnvVar>$(PATH)</EnvVar>
+                                         <LocalAppData>Overwritten!</LocalAppData>
```

Comment:
LocalAppData does not exist on Unix. Use TestEnvironment.SetEnvironmentVariable to set test env vars before the evaluation instead of using system defined env vars like Path or LocalAppData. Between different OSes and in CI environments it gets quite shaky to use predefined system env vars.

Reply:
Updated with latest commit.